services/provider/server/server.go

```diff
@@ -418,6 +426,21 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 
 	}
 
+	//Enable CSI-OMAP is storageClusterPeer is present
+	clientMappingConfig, err := s.getClientMapping(ctx)
+	if err != nil {
+		return nil, err
+	}
+	_, hasClientID := clientMappingConfig.Data[string(consumerResource.UID)]
```

Comment:
This is not good enough, what if the mapping is broken? For example the key exists but not a value
I would suggest doing this instead

```suggestion
	clientIsMapped := clientMappingConfig.Data[string(consumerResource.UID)] != ""
```